### PR TITLE
Add partition_time as Unix epoch partition to enable easier query format across month/year

### PR DESCRIPTION
**Pull request by serbantanasa**
## Background

Adds partition_time to athena to make indexed queries less horrible. This will be needed for scheduled queries with time windows.

## Changes

1. Add new index partition_time, unix time bigint rounded to the hour. 
2. Add index on partition_time partition
3. Update table update code to add the extra partition 
4. Update glue sync code to handle upgrade from 4 partitions to 5
5. Update unit and integration tests to account for new structure expected. 

## Testing

- Passes unit tests
- Passes integration tests
- New deploy is clean
- Passing upgrade tests

### CODE REVIEW

File: internal/log_analysis/awsglue/glue_timebin.go [hunk 1]
```diff
@@ -256,6 +261,8 @@ func PartitionTimeFromValues(values []*string) (tm time.Time, err error) {
 		tm = unpackValues(values[0], values[1], values[2], nil)
 	case 4:
```
Comment (rleighton) [inline internal/log_analysis/awsglue/glue_timebin.go:262]: 4,5 and remove case for 5

File: internal/log_analysis/awsglue/glue_timebin.go [hunk 2]
```diff
@@ -160,6 +161,10 @@ func TimebinFromTable(tbl *glue.TableData) (GlueTableTimebin, error) {
 		if keyNames[0] == "year" && keyNames[1] == "month" && keyNames[2] == "day" && keyNames[3] == "hour" {
 			return GlueTableHourly, nil
 		}
+	case 5:
```
Comment (rleighton) [inline internal/log_analysis/awsglue/glue_timebin.go:164]: You can just make the above 4,5 and remove this case

File: internal/log_analysis/gluetasks/sync.go [hunk 3]
```diff
@@ -203,6 +204,7 @@ func (s *SyncTablePartitions) syncTable(ctx context.Context, api glueiface.GlueA
 			numSynced, err := processPartitionUpdates(ctx, api, tasks, s.NumWorkers)
 			s.Stats.NumSynced += int(numSynced)
 			if err != nil {
+				log.Debug("error caused interrupt", zap.Error(err))
```
Comment (rleighton) [inline internal/log_analysis/gluetasks/sync.go:207]: remove , we are trying to avoid deep logging

**Review by rleighton — COMMENTED** [review]
Be sure to add me to the sync PR in enterprise, we need to make 1 change in snowflake for the views to skip `partition_time` like we skip year, month, day, hour.

After addressing the comments, open up the PR.

File: internal/log_analysis/awsglue/glue_timebin.go (hunk 1 above)
Comment (serbantanasa) [reply internal/log_analysis/awsglue/glue_timebin.go]: done

File: internal/log_analysis/athenaviews/athenaviews_test.go [hunk 4]
```diff
@@ -45,9 +45,9 @@ func TestGenerateViewAllLogs(t *testing.T) {
 	table2 := awsglue.NewGlueTableMetadata(pantherdb.LogProcessingDatabase, "table2", "test table2", awsglue.GlueTableHourly, &table2Event{})
 	// nolint (lll)
 	expectedSQL := `create or replace view panther_views.all_logs as
-select 'panther_logs' AS p_db_name,day,hour,month,NULL AS p_any_aws_account_ids,NULL AS p_any_aws_arns,NULL AS p_any_aws_instance_ids,NULL AS p_any_aws_tags,p_any_domain_names,p_any_ip_addresses,p_any_md5_hashes,p_any_sha1_hashes,p_any_sha256_hashes,p_event_time,p_log_type,p_parse_time,p_row_id,p_source_id,p_source_label,year from panther_logs.table1
+select 'panther_logs' AS p_db_name,day,hour,month,NULL AS p_any_aws_account_ids,NULL AS p_any_aws_arns,NULL AS p_any_aws_instance_ids,NULL AS p_any_aws_tags,p_any_domain_names,p_any_ip_addresses,p_any_md5_hashes,p_any_sha1_hashes,p_any_sha256_hashes,p_event_time,p_log_type,p_parse_time,p_row_id,p_source_id,p_source_label,partition_time,year from panther_logs.table1
```
Comment (kostaspap) [inline internal/log_analysis/athenaviews/athenaviews_test.go:48]: nitpick: what about moving `partition_time` and `year` in the top together with the other partitions? 
`select 'panther_logs' AS hour, day, month, year, partition_time, p_db_name,....`

File: internal/log_analysis/awsglue/partition_test.go [hunk 5]
```diff
@@ -85,7 +89,12 @@ func TestCreatePartitionFromS3Log(t *testing.T) {
 			Key:   "hour",
 			Value: "15",
 		},
+		{
+			Key:   "partition_time",
+			Value: "1582729200",
+		},
 	}
+	//awsglue.PartitionColumnInfo{Key:"partition_time", Value:"1582729200"}
```
Comment (kostaspap) [inline internal/log_analysis/awsglue/partition_test.go:97]: you can remove comment.

File: internal/log_analysis/awsglue/table_metadata.go [hunk 6]
```diff
@@ -270,6 +271,12 @@ func (gm *GlueTableMetadata) CreateOrUpdateTable(glueClient glueiface.GlueAPI, b
 					aws.String("day"),
 				},
 			},
+			{
+				IndexName: aws.String("hour_idx"),
```
Comment (kostaspap) [inline internal/log_analysis/awsglue/table_metadata.go:275]: better name it differently e.g. `partition_time_idx`? Following the pattern from the above two, I would expect an `hour_idx` to be 
```
Keys: []*string{
  aws.String("year"),
  aws.String("month"), 
  aws.String("day"), 
  aws.String("hour"),
}
```

Comment (kostaspap) [inline internal/log_analysis/awsglue/table_metadata.go:275]: This needs to be added in the `CreateTableIfNotExists` method as well

Comment (serbantanasa) [reply internal/log_analysis/awsglue/table_metadata.go]: added

Comment (serbantanasa) [reply internal/log_analysis/awsglue/table_metadata.go]: good point

File: internal/log_analysis/awsglue/partition_test.go (hunk 5 above)
Comment (serbantanasa) [reply internal/log_analysis/awsglue/partition_test.go]: done

**Review by kostaspap — APPROVED** [review]
LGTM given that this has been tested E2E

**Pull request merged**